adapters/admatic/admatic.go

```diff
@@ -0,0 +1,155 @@
+package admatic
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"text/template"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/macros"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint *template.Template
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	template, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+
+	bidder := &adapter{
+		endpoint: template,
+	}
+
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	Host := ""
+
+	for _, imp := range request.Imp {
+		admaticExt, err := getImpressionExt(imp)
+		if err != nil {
+			return nil, []error{err}
+		}
+		if admaticExt.Host != "" {
+			if Host == "" {
+				Host = admaticExt.Host
+			} else if Host != admaticExt.Host {
+				return nil, []error{&errortypes.BadInput{
+					Message: "There must be only one Hos",
+				}}
+			}
+		} else {
+			return nil, []error{&errortypes.BadInput{
+				Message: "The Hos must not be empty",
+			}}
+		}
+	}

```

Comment:
`host` is defined as required param in `static/bidder-params/admatic.json` 

Prebid server core framework will ensure that admaticExt.Host is not empty. Therefore, `if admaticExt.Host != "" {` check can be removed

Reply:
@bakicam based on adapter code, expectation here is  that all imps should have same host value. 

Should explictily mentioned this behaviour in bidder docs so that publishers are aware of this requirement,